upload_Faces_Batch.py

```diff
@@ -0,0 +1,119 @@
+import requests
+import argparse
+import sys
+import json
+import csv
+import time
+
+class uploadDirectory:
+    #Set up workspace for API calls to Rhombus Systems
+    def __init__(self,args):
+        #Initialize argument parser
+        self.argParser = self.__initArgParser()
+        self.args = self.argParser.parse_args(args)
+        #Create a session to set default call Header information
+        self.session = requests.session()
+        self.session.headers = {"x-auth-scheme": "api-token","x-auth-apikey": self.args.apiKey}
+        #Set a default base URL for api calls
+        self.url = "https://api2.rhombussystems.com/api/"
+
+    #Define arguments which the user may be prompted for
+    @staticmethod
+    def __initArgParser():
+        argParser = argparse.ArgumentParser(description='Upload a selection of faces for facial recognition. List file names in a CSV on individual lines.Please follow the nameing protocal: First-Last_#.jpg')
+        argParser.add_argument("apiKey", help="Get this from your Console.")
+        argParser.add_argument("addr", help="Local Address of Photo Directory CSV")
+        argParser.add_argument("-time","-t", help="Include Timestamp in Report title", action='store_true')
+        return argParser
+
+    def execute(self):
+        ##Upload Faces to prelaod facial ID
+        #Parse through CSV or Folder
+        uploadedFaces = []
+        try:
+            with open(self.args.addr) as csvInput:
+                inputReader = csv.reader(csvInput)
+                for row in inputReader:
+                    faceAddr = str(row[0])#Address of face
+                    fileName = faceAddr.split("/")
+                    fileName = fileName[len(fileName)-1]#Filename of face
+                    try:
+                        #Attempt upload of face
+                        files = {"files[]": (fileName, open(faceAddr, "rb"), "image/jpeg")}
+                        uploadResponse = self.session.post(self.url+'upload/faces',files=files)
+                        #Report unexpected status codes
+                        if(uploadResponse.status_code != 201):
+                            print("Error encountered while uploading face:",faceAddr,"(/api/upload/faces)")
+                            continue
+                        #Store name and filename for verification            
+                        uploadedFaces.append(fileName)
+                    except FileNotFoundError:#Catch non valid faces, and continue
+                        print(fileName,"is not a valid jpg.")
+                        continue
+        except FileNotFoundError:
+            print("FileNotFoundError: Indicated CSV file is non existent.")
+            return
+
+        #Pause to allow faces to be processed
+        time.sleep(30)
```

Comment:
Probably better here to poll this endpoint every couple seconds and wait until either all of the uploaded faces have entries, or until a timeout period specified by cli argument (default to something long like 5 min).  While in general the indexing of a face occurs very quickly, it could take a while if a lot of faces are uploaded close together.

Reply:
That makes sense, newest commit polls until elapsed timeout or all uploaded faces are accounted for.